File: Algorithms/Brute_force_algorithms/maximum_sum.py
# Given an array of integers, find the maximum sum of a subarray of consecutive elements

def max_subarray_sum(arr):
    max_sum = arr[0]
    
    for i in range(len(arr)):
        current_sum = 0
        for j in range(i, len(arr)):
            current_sum += arr[j]
            if current_sum > max_sum:
                max_sum = current_sum
    
    return max_sum

arr = [1, -3, 4, 2, -1, 6, -5]
max_sum = max_subarray_sum(arr)
print("Maximum sum of subarray:", max_sum)

File: Algorithms/Brute_force_algorithms/maximum_sum.c
// Given an array of integers, find the maximum sum of a subarray of consecutive elements

#include <stdio.h>

int maxSubarraySum(int arr[], int size) {
    int maxSum = arr[0];
    
    for (int i = 0; i < size; i++) {
        int currentSum = 0;
        for (int j = i; j < size; j++) {
            currentSum += arr[j];
            if (currentSum > maxSum) {
                maxSum = currentSum;
            }
        }
    }
    
    return maxSum;
}

int main() {
    int arr[] = {1, -3, 4, 2, -1, 6, -5};
    int size = sizeof(arr) / sizeof(arr[0]);
    
    int maxSum = maxSubarraySum(arr, size);
    
    printf("Maximum sum of subarray: %d\n", maxSum);
    printf("\n");
    
    return 0;
}
